### ISSUE TITLE
Needs tests

### ISSUE THREAD

**Issue by jayvdb** [carljm/pytest-smartcov#4]
I am concerned about switching from pytest-cov to pytest-smartcov because this doesnt have any tests, so it is hard to know whether it is compatible with new versions of pytest & coverage as they arrive.

**Comment by carljm** [comment]
Fair point! Would definitely be good to add tests. There’s a reasonable chance it does need updating for newer pytest and coverage; I haven’t had the opportunity to use pytest or this project for a couple years. 

Given that I won’t likely be able to invest much in adding tests or otherwise updating the project in the near term, but I’d be happy to review PRs and also to add maintainers with commit access, given a solid contribution or two.

**Comment by jayvdb** [comment]
I've managed to solve my own immediate scratch with https://github.com/krkd/pytest-cov-threshold and some glue provided by https://github.com/jayvdb/tox-backticks to dynamically create its config file `.threshold.json`.

However the [issues there](https://github.com/krkd/pytest-cov-threshold) include some rather critical problems (and iirc its own test suite didnt pass either), and there has been no response so far.  I do like its approach, leaveraging off pytest-cov instead of trying to compete with it.  Even if those are fixed, pytest-smartcov 's dynamic resolution is an important feature I need long term, as I dont want to require tox in order to run pytest.

In my new tox-backticks repo is an effective way to quickly create basic tests for tools that require isolated environments and need to be tested against lots of versions of dependencies, with the tests all written in a single `tox.ini` using command `python -c 'assert ...'` to check the state within the isolated environment.  I am fairly sure I could create a similar test suite for pytest-smartcov quite quickly if that is acceptable to you as an interim solution.

This approach is not so good for verifying what happens when a expected condition fails, as the tox error handling takes over. But it is good for quickly establishing some minimal acceptance criteria to prove it works correctly when used correctly, and thus to identify when it is incompatible with a versions of a dependency.

**Comment by carljm** [comment]
Any test suite would be preferable to none!
